robustirc/src/jepsen/robustirc.clj

```diff
@@ -0,0 +1,230 @@
+(ns jepsen.robustirc
+  (:require
+    [clojure.tools.logging    :refer [debug info warn]]
+    [clojure.java.io          :as io]
+    [clojure.string           :as str]
+    [clojure.core.async       :as async]
+    [clj-http.client :as httpclient]
+    [digest :as digest]
+    [jepsen
+      [client :as client]
+      [core :as jepsen]
+      [db :as db]
+      [tests :as tests]
+      [control :as c :refer [|]]
+      [checker :as checker]
+      [nemesis :as nemesis]
+      [generator :as gen]]
+    [jepsen.os.debian :as debian]
+    [cheshire.core            :as json])
+  (:import
+    [java.net URL]))
+
+(defn db [version]
+  "Rethinkdb (ignores version)."
+  (reify db/DB
+    (setup! [this test node]
+      (c/su
+        (c/ssh* {:cmd "killall robustirc"})
+	(try (c/exec :dpkg-query :-l :golang-go)
+	     (catch RuntimeException _
+	       (info "Installing golang-go")
+	       (c/exec :apt-get :install :-y :golang-go)))
+	(try (c/exec :dpkg-query :-l :mercurial)
+	     (catch RuntimeException _
+	       (info "Installing mercurial")
+	       (c/exec :apt-get :install :-y :mercurial)))
+	(c/exec :env "GOPATH=~/gocode"
+	        :go :get :-u "github.com/robustirc/robustirc")
+
+        ; generated by resources/gencert.go
+        (c/upload (.getFile (io/resource "cert.pem")) "/tmp/cert.pem")
+        (c/upload (.getFile (io/resource "key.pem")) "/tmp/key.pem")
+        (c/ssh* {:cmd "rm -rf /var/lib/robustirc"})
+        (c/ssh* {:cmd "mkdir -p /var/lib/robustirc"})
+        (jepsen/synchronize test)
+
+        (let [cmd (str
+                "/sbin/start-stop-daemon --start --background --exec ~/gocode/bin/robustirc --"
+                " -listen=" (name node) ":13001"
+                " -network_password=secret"
+                " -network_name=jepsen"
+                " -tls_cert_path=/tmp/cert.pem"
+                " -tls_ca_file=/tmp/cert.pem"
+                " -tls_key_path=/tmp/key.pem"
+                " -singlenode")]
+          (if-not (= node (jepsen/primary test))
+                (Thread/sleep 1000)
+                (do
+                  (info node (str "running: " cmd))
+                  (c/ssh* {:cmd cmd})
+                  (Thread/sleep 5000))))
+        (jepsen/synchronize test)
+        (let [cmd (str
+                "/sbin/start-stop-daemon --start --background --exec ~/gocode/bin/robustirc --"
+                " -listen=" (name node) ":13001"
+                " -network_password=secret"
+                " -network_name=jepsen"
+                " -tls_cert_path=/tmp/cert.pem"
+                " -tls_ca_file=/tmp/cert.pem"
+                " -tls_key_path=/tmp/key.pem"
+                " -join=" (name (jepsen/primary test)) ":13001")]
+          (if (= node (jepsen/primary test))
+                (Thread/sleep 100)
+                (do
+                  (info node (str "running: " cmd))
+                  (c/ssh* {:cmd cmd})
```

Comment:
You're welcome to use strings, symbols, or keywords; they all work the same in this api. :foo is just a little shorter to type.